lib/files.go

```diff
@@ -0,0 +1,194 @@
+package lib
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// AppendIfMissing appends the content string to the given file
+func AppendIfMissing(filePath string, content string) error {
+	fileToModify, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	Check(err)
+
+	fileContents, err := ioutil.ReadFile(fileToModify.Name())
+	Check(err)
+
+	if !strings.Contains(string(fileContents), content) {
+		_, err = fileToModify.Write([]byte(content))
+	}
+
+	fileToModify.Close()
+
+	return err
+}
+
+// RenameDir renames a directory while handling cross-link issues
+// credit https://github.com/rawlingsj/jx/blob/master/pkg/util/files.go
+func RenameDir(src string, dst string, force bool) (err error) {
```

Comment:
If you want this to have the same semantics as rename, I think you might want to make a temporary directory in the same directory as the target, copy into the temporary target, and then do a rename from the temporary path to the target path.